content/300-guides/100-performance-and-optimization/150-connection-management/index.mdx

```diff
@@ -0,0 +1,286 @@
+  prisma = global.prisma
+}
+
+export default prisma
+```
+
+The way that you import and use the client does not change:
+
+```ts file=app.ts
+import prisma from './client'
+
+async function main() {
+    const allUsers = await prisma.user.findMany();
+}
+
+main();
+```
+
+## Serverless environments (FaaS)
+
+Examples of serverless environments include Node.js functions hosted on AWS Lambda. Use the following checklist as a guide to connection management in serverless environments:
+
+- ☐ [Set connection limit](#recommended-connection-limit-1) based on whether you have an external connection pooler
+- ☐ Optionally [tune the pool size](#optimizing-the-connection-pool) if you are using an external connection pooler
+- ☐ [Instantiate `PrismaClient` outside the handler](#instantiate-prismaclient-outside-the-handler) and do not explicitly `$disconnect()`
+- ☐ Configure [function concurrency](#concurrency-limits) and handle [idle connections](#zombie-connections)
+
+### The serverless challenge
+
+In a serverless environment, each function creates **its own instance** of `PrismaClient`, and each client instance has its own connection pool. 
+
+Consider the following example, where a single AWS Lambda function uses `PrismClient` to connect to a database. The `connection_limit` is **3**:
+
+![An AWS Lambda function connecting to a database.](./serverless-connections.png)
+
+A traffic spike causes AWS Lambda to spawn two additional lambdas to handle the increased load. Each lambda creates an instance of `PrismaClient`, each with a `connection_limit` of **3**, which results in a maximum of **9** connections to the database:
+
+![Three AWS Lambda function connecting to a database.](./serverless-connections-2.png)
+ 
+200 _concurrent functions_ (and therefore 600 possible connections) responding to a traffic spike 📈 can exhaust the database connection limit very quickly.
+
+1. Start by [setting the `connection_limit` is set to `1`](#recommended-connection-limit-1)
+2. If a lower connection limit is not enough, consider using an [external connection pooler like PgBouncer](#external-connection-poolers)
+
+### Recommended connection limit
+
+The recommended connection limit in serverless environments depends on whether you are using an [external connection pooler](#external-connection-poolers).
+
+#### Without an external connection pooler
+
+We recommend that you set the `connection_limit` to **1** if you are **not** using an external connection pooler. Each incoming request starts a short-lived Node.js process, and many concurrent functions with a high `connection_limit` can quickly **exhaust the _database_ connection limit** during a traffic spike.
+
+The following example demonstrates how to set the `connection_limit` to 1 in your connection string:
+
+<TabbedContent tabs={[<FileWithIcon text="PostgreSQL" icon="database"/>, <FileWithIcon text="MySQL" icon="database"/>]}>
+<tab>
+
+  ```
+  postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
```

Comment:
missing the schema

Reply:
@janpio Why is specifying the schema important here? Isn't `schema=public` the default?